src/azure-cli-core/azure/cli/core/commands/sensitive.py

```diff
@@ -0,0 +1,85 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from knack.util import StatusTag
+
+_SENSITIVE_TAG = '[Sensitive]'
+_sensitive_kwarg = 'sensitive_info'
+_config_key = 'sensitive'
+
+
+def resolve_sensitive_info(cli_ctx, name):
+
+    def _get_command(name):
+        return cli_ctx.invocation.commands_loader.command_table[name]
+
+    def _get_command_group(name):
+        return cli_ctx.invocation.commands_loader.command_group_table.get(name, None)
+
+    sensitive_info = None
+    try:
+        command = _get_command(name)
+        sensitive_info = getattr(command, _sensitive_kwarg, None)
+    except KeyError:
+        command_group = _get_command_group(name)
+        group_kwargs = getattr(command_group, 'group_kwargs', None)
+        if group_kwargs:
+            sensitive_info = group_kwargs.get(_sensitive_kwarg, None)
+    return sensitive_info
+
+
+# pylint: disable=too-many-instance-attributes
+class SensitiveItem(StatusTag):
```

Comment:
I still keep this new tag even though there's no command adding sensitive tag for now.

With this new tag, we have the ability to manually mark some properties as sensitive for specific command. For example, `az keyvault secret set` command will return `value` property which should be sensitive but may not be recognized by regexes.

Reply:
If we apply credential detection/redaction on _all_ commands, this `sensitive_info` flag won't be needed at all.